web/params.go

```diff
@@ -355,6 +357,29 @@ func ParamsFromRequest(r *http.Request) *Params {
 		params.ExcludePolicyConstrained = val
 	}
 
+	if val, ok := props["group_source"]; ok {
+		switch val {
+		case "custom":
+			params.GroupSource = model.GroupSourceCustom
+		case "ldap":
+			params.GroupSource = model.GroupSourceLdap
+		default:
+			params.GroupSource = model.GroupSourceLdap
+		}
+	}
+	if val := query.Get("group_source"); val != "" {
+		switch val {
+		case "custom":
+			params.GroupSource = model.GroupSourceCustom
+		case "ldap":
+			params.GroupSource = model.GroupSourceLdap
+		default:
+			params.GroupSource = model.GroupSourceLdap
+		}
+	}
```

Comment:
Is the purpose of these to do some early validation or just setting the request param? I don't see them used anywhere.

Reply:
It was added for use at https://github.com/mattermost/mattermost-server/pull/18839/files#diff-060eb667249b07159dcd89b520fa51777abf5bd355241d2e5c7e6bb58a93d367R852, but to me there's no need for both `props` and `query` retrieval, so I removed the props one. Also it was missing a test, which I added and that exposed that it was missing from the Go client4 too. Updated.